pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/route/TriggerSomaticResultSurveyRoute.java

```diff
@@ -0,0 +1,149 @@
+package org.broadinstitute.dsm.route;
+
+import com.google.gson.JsonObject;
+import com.google.gson.JsonParser;
+import lombok.AllArgsConstructor;
+import lombok.Data;
+import lombok.NonNull;
+import org.apache.commons.lang3.StringUtils;
+import org.broadinstitute.dsm.db.DDPInstance;
+import org.broadinstitute.dsm.db.SomaticResultUpload;
+import org.broadinstitute.dsm.db.SurveyTrigger;
+import org.broadinstitute.dsm.exception.AuthorizationException;
+import org.broadinstitute.dsm.exception.DSMBadRequestException;
+import org.broadinstitute.dsm.exception.DsmInternalError;
+import org.broadinstitute.dsm.security.RequestHandler;
+import org.broadinstitute.dsm.service.SomaticResultUploadService;
+import org.broadinstitute.dsm.statics.ESObjectConstants;
+import org.broadinstitute.dsm.statics.RoutePath;
+import org.broadinstitute.dsm.util.DDPRequestUtil;
+import org.broadinstitute.dsm.util.UserUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import spark.QueryParamsMap;
+import spark.Request;
+import spark.Response;
+
+public class TriggerSomaticResultSurveyRoute extends RequestHandler {
+    private final SomaticResultUploadService service;
+    private static final Logger logger = LoggerFactory.getLogger(TriggerSomaticResultSurveyRoute.class);
+    protected static final String REQUIRED_SURVEY_CREATION_ROLE = "survey_creation";
+    protected static final String REQUIRED_VIEW_ROLE = "view_shared_learnings";
+    protected static final String REQUIRED_MANIPULATE_ROLE = "upload_ror_file";
+
+    public TriggerSomaticResultSurveyRoute(SomaticResultUploadService service) {
+        this.service = service;
+    }
+
+    @Override
+    protected Object processRequest(Request request, Response response, String userId) throws Exception {
+        QueryParamsMap queryParams = request.queryMap();
+        String realm = queryParams.get(RoutePath.REALM).value();
+        if (StringUtils.isBlank(realm)) {
+            throw new DSMBadRequestException("realm is a required query parameter.");
+        }
+        if (isAuthorized(userId, realm)) {
```

Comment:
Just to be clear, do you find this easier to read?
if (!isAuthorized(userId, realm) {
   throw new AuthorizationException();
} else {
     ....do work
}

..... or do you prefer:
if (!isAuthorized(userId, realm) {
    throw new AuthorizationException();
}
....do work

I'm also trying to understand what you mean by this:
"As is, the outcome of the conditionals are removed from the condition test."

Reply:
First, to be clear, this is mostly a nit, but I do have a problem with other areas of DSM where this style creates serious readability problems. Specifically, the disposition of the conditional, particularly when it is terminal, is effectively hidden by many lines of code. Here there are just a handful of lines, so no biggie.
In my mind, there is not really an else here -- since the else just says 'done', and the other side of the else is 'continue'. Not surprisingly I like the latter code styes you proposed. It makes the main line of the code clearer by keeping it at the margin.
Now dismounting the soapbox. :-)